include/gunrock/framework/operators/filter/predicated.hxx

```diff
@@ -82,46 +83,43 @@ void execute_mgpu(graph_t& G,
   input_ends[num_gpus - 1] = input->end();
   
   // Map
-  size_t new_sizes[num_gpus];
+  int new_sizes[num_gpus];
   for(int i = 0; i < num_gpus; i++) {
     auto ctx = context.get_context(i);
     cudaSetDevice(ctx->ordinal());
     
-    auto new_length = thrust::copy_if(
+    auto new_end = thrust::copy_if(
       thrust::cuda::par.on(ctx->stream()),
```

Comment:
This is great! Now, if the stream is non-blocking, this call should be non-blocking as well (the change i mentioned above)